### ISSUE TITLE
duplicates conflict - play-games-plugin-for-unity

### ISSUE THREAD

**Issue by mgrogin** [firebase/quickstart-unity#716]
### Please fill in the following fields:
Unity editor version: 2019.2.17f1
Firebase Unity SDK version: 6.15.0
Source you installed the SDK (.unitypackage or Unity Package Manager): Unity Package Manager
Firebase plugins in use (Auth, Database, etc.): App (core), Authentication, Cloud Messaging, Crashlytics, Dynamic Links, Realtime Database, Remote Config, Analytics for firebase
Additional SDKs you are using (Facebook, AdMob, etc.): Facebook, IronSource, play-games-plugin-for-unity, External Dependency Manager for Unity 1.2.155
Platform you are using the Unity editor on (Mac, Windows, or Linux): Mac
Platform you are targeting (iOS, Android, and/or desktop): Android, Ios
Scripting Runtime (Mono, and/or IL2CPP): IL2CPP

### Please describe the issue here:
(Please list the full steps to reproduce the issue. Include device logs, Unity logs, and stack traces if available.)
Since I installed the ironsource sdk which uses the External Dependency Manager for Unity I installed Firebase the same way with the latest version 1.2.155.
I was having trouble building for android due to duplicates with different versions of com.google.firebase:firebase-analytics and androidPackage spec="com.google.firebase:firebase-messaging:20.1.7

the way I was able to solve it was by changing the firebase  dependencies xml in /Library/PackageCache/

changed `<androidPackage spec="com.google.firebase:firebase-analytics:17.4.1">` to `<androidPackage spec="com.google.firebase:firebase-analytics:17.4.3">`

in:
/Library/PackageCache/com.google.firebase.remote-config@6.15.0/Firebase/Editor/RemoteConfigDependencies.xml
/Library/PackageCache/com.google.firebase.dynamic-links@6.15.0/Firebase/Editor/DynamicLinksDependencies.xml
/Library/PackageCache/com.google.firebase.messaging@6.15.0/Firebase/Editor/MessagingDependencies.xml
/Library/PackageCache/com.google.firebase.database@6.15.0/Firebase/Editor/DatabaseDependencies.xml
/Library/PackageCache/com.google.firebase.app@6.15.0/Firebase/Editor/AppDependencies.xml
/Library/PackageCache/com.google.firebase.crashlytics@6.15.0/Firebase/Editor/CrashlyticsDependencies.xml
/Library/PackageCache/com.google.firebase.analytics@6.15.0/Firebase/Editor/AnalyticsDependencies.xml
/Library/PackageCache/com.google.firebase.auth@6.15.0/Firebase/Editor/AuthDependencies.xml

changed `<androidPackage spec="com.google.firebase:firebase-messaging:20.1.7">` to `<androidPackage spec="com.google.firebase:firebase-messaging:20.2.0">`
in:
/Library/PackageCache/com.google.firebase.messaging@6.15.0/Firebase/Editor/MessagingDependencies.xml

I think the conflict was coming from 
`<androidPackage spec="com.google.games:gpgs-plugin-support:0.10.06">`
in Assets/GooglePlayGames/Editor/GooglePlayGamesPluginDependencies.xml
but Im not sure...

Is there a better way to address this? It feels wrong to change something in a cache folder...
Thanks.
### Please answer the following, if applicable:
Have you been able to reproduce this issue with just the Firebase Unity quickstarts (this GitHub project)?

What's the issue repro rate? (eg 100%, 1/5 etc)

**Comment by patm1987** [comment]
It is ideal to not have to poke around in the caches directory (this should be invisible).

It would be really helpful if you could get us some minimum viable steps to reproduce [on a quickstart](https://github.com/firebase/quickstart-unity). For example: do you see the issue on the [Messaging Quickstart](https://github.com/firebase/quickstart-unity/tree/master/messaging/testapp) after adding the [Play Games Plugin](https://github.com/playgameservices/play-games-plugin-for-unity)?

**Comment by chkuang-g** [comment]
@mgrogin 

First of all, what is the error message in the log?

What were you trying to change in `RemoteConfigDependencies.xml` and `MessagingDependencies.xml`?  Your post seem missing some information.

Besides, I am not familiar with ironsource SDK and had a hard time to get one.  Do they ship with EDM4U and/or `Dependencies.xml`?  Could you posts its content?

I did installed a couple of Firebase SDK through UPM and Google Play Game plugin and I can resolve 
 dependencies and build apk with no problem.  It will be helpful if you can provide a minimum viable project to reproduce the isse.

One last note: Since you install Firebase through Unity Package Manager, and Google Play Game plugin is only available as `.unitypackage` (and I assume IronSource SDK as well), please make sure to delete `Assets/ExternalDependencyManager` or `Assets/PlayServicesResolver` folder after installing them.

**Comment by mgrogin** [comment]
Hmm Im not sure how this line got erased.
What I changed was:
<androidPackage spec="com.google.firebase:firebase-analytics:17.4.1"> to <androidPackage spec="com.google.firebase:firebase-analytics:17.4.3">
in all the dependencies xnls I mentioned
<androidPackage spec="com.google.firebase:firebase-messaging:20.1.7"> to <androidPackage spec="com.google.firebase:firebase-messaging:20.2.0">
in MessagingDependencies.xml

This is the ironsource sdk https://developers.ironsrc.com/ironsource-mobile/unity/unity-plugin/
they ship with EDM4U but I didn't import it because it wasn't the latest version

**Comment by chkuang-g** [comment]
@mgrogin 

TL;DR; I would recommend you to revert your change in Dependencies.xml, remove all `aar` and `.jar` files under `Assets/Plugins/Android` folder, and do Force Resolve again.

Here is a longer explanation
- I installed Firebase Analytics, Remote Config 6.15.0 through UPM, Google Play Game through `.unitypackage` and IronSource through `.unitypackage`.  Did Force Resolve and made an apk.  Did not see any error log.  It would be very helpful if you can provide the log.  I would recommend you to turn on `Verbose Logging` in the settings of `Android Resolver` for more information.
- It is never a good idea to bump up Firebase Android SDK version since we only run and test Unity SDK against the version specified in Dependencies.xml.  It is not a good idea to modify it under `Library/PackageCache` folder either since Unity has some tendency to revert your changes.  I cannot figure out a reason why you need to do so.  Neither Google Play Game nor IronSource SDK depends on Firebase Android SDK. I suspect some `.aar` files were unlabelled so that Android Resolver no longer able to manage them, and result in duplicated symbol. The resolution is to remove them and do Force Resolve again.
- While I did not see any conflict between IronSource and Google Play Game SDK, if there is any, I would recommend you to file a support ticket to them.

Hope this help!
Shawn

**Comment by mgrogin** [comment]
I'm still working out the issue but I think I've resolved it.
There must've been leftovers somewhere.
I think the issue was cause by upgrading from a manual Firebase  to using the External Dependency Manager .
My approach now was to make sure everything  firebase was deleted before installing External Dependency Manager and then adding it from the package manager instead of letting External Dependency Manager migrate it.
You're comment has been very helpful Shawn ( @chkuang-g )
I'll post an update if I find anything else

**Comment by alexames** [comment]
Looks like your issue was fixed. Since this hasn't had any updates in a while I'm going to go ahead and close it. If you have an update feel free to reopen this issue (if it's related, obviously) or open a new issue.
